core/types/block.go

```diff
@@ -165,6 +165,23 @@ func (h *Header) EmptyReceipts() bool {
 	return h.ReceiptHash == EmptyReceiptsHash
 }
 
+// CheckTransactionConditional validates the block preconditions against the header
+func (h *Header) CheckTransactionConditional(cond *TransactionConditional) error {
+	if cond.BlockNumberMin != nil && cond.BlockNumberMin.Cmp(h.Number) > 0 {
+		return fmt.Errorf("failed block number minimum constraint")
+	}
+	if cond.BlockNumberMax != nil && cond.BlockNumberMax.Cmp(h.Number) < 0 {
+		return fmt.Errorf("failed block number maximmum constraint")
+	}
```

Comment:
If BlockNumberMax == h.Number, then the comparison equals 0, and it's included. Is the upper bound meant to be inclusive? I'd expect an exclusive upper bound.

Reply:
Per the [spec](https://notes.ethereum.org/@yoav/SkaX2lS9j) "blockNumberMax: [optional] maximum block number for inclusion"

I interpreted this as inclusive bounds (same working with timestamp bounds)